tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -6,72 +6,76 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
-	sessionaws "github.com/aws/aws-sdk-go/aws/session"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/iam"
-	v1 "github.com/openshift/api/config/v1"
-	client "github.com/openshift/client-go/config/clientset/versioned"
 	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/openshift"
 )
 
-// svc is a AWS specific struct which contains service information needed to communicate with AWS and the existing cluster
-// svc should remain private since secretes should not be exposed
-type svc struct {
-	// service account for AWS EC2
+// cloudProviderInfo is an provider specific struct which contains clients for EC2, IAM,
+// and the existing OpenShift cluster that is running on EC2.
+// This is an implementation of the Cloud interface.
+type cloudProviderInfo struct {
+	// A client for EC2.
 	EC2 *ec2.EC2
-	// service account for AWS IAM
+	// A client for IAM.
 	IAM *iam.IAM
-	// client of the existing cluster
-	clientset *client.Clientset
-	// infraID is the readable unique infrastructure name generated by openshift-install for new infrastructures
-	infraID *string
-	// dir is the path to where created instance and security group information is stored
-	dir *string
+	// clientset is the client of the existing OpenShift cluster.
+	Clientset *openshift.Client
+	// resourceTrackerDir is the path to where created instance and security group information is stored.
+	resourceTrackerDir string
 }
 
-// New is the factory of AWS Cloud interface
-// It takes the generic type of information and returns an interface containing information specific to AWS
-func New(clientset *client.Clientset, infrastructure *v1.Infrastructure, credentialPath, credAccount, dir *string) (cloudprovider.Cloud, error) {
-	region := infrastructure.Status.PlatformStatus.AWS.Region
-	infraId := infrastructure.Status.InfrastructureName
-	session, err := NewSession(credentialPath, credAccount, &region)
+// New takes the generic type of information and returns an interface containing cloudProviderInfo specific to AWS.
```

Comment:
How is the information generic? There is info specific to AWS. I think you can just say what the function does given you are describing each input.